kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -215,6 +207,35 @@ public CompiledDefinition run(File definitionFile, String mainModuleName, String
         return def;
     }
 
+    private void buildBisonParser(Sort sort, String modName, String name, Module mod) {
+        String filename = "parser_" + sort.name() + "_" + modName;
+        File outputFile = files.resolveKompiled(filename);
+        File linkFile = files.resolveKompiled("parser_" + name);
+        new KRead(kem, files, InputModes.PROGRAM, globalOptions).createBisonParser(mod, sort, outputFile, kompileOptions.genGlrBisonParser, null, kompileOptions.bisonStackMaxDepth);
+        try {
+            linkFile.delete();
+            Files.createSymbolicLink(linkFile.toPath(), files.resolveKompiled(".").toPath().relativize(outputFile.toPath()));
+        } catch (IOException e) {
+            throw KEMException.internalError("Cannot write to kompiled directory.", e);
+        }
+    }
+
+    private Definition postProcessJSON(Definition defn, String postProcess) {
+        List<String> command = new ArrayList<>(Arrays.asList(postProcess.split(" ")));
+        Map<String, String> environment = new HashMap<>();
+        try {
+            String inputDefinition = new String(ToJson.apply(defn), "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw KEMException.criticalError("Could not encode definition to JSON!");
+        }
+        RunProcess.ProcessOutput output = RunProcess.execute(environment, files.getProcessBuilder(), command.toArray(new String[command.size()]));
+        if (output.exitCode != 0) {
+            throw KEMException.criticalError("Post-processing returned a non-zero exit code: "
+                    + output.exitCode + "\nStdout:\n" + new String(output.stdout) + "\nStderr:\n" + new String(output.stderr));
+        }
+        return JsonParser.parseDefinition(output.stdout);
+    }
```

Comment:
I see two ways to send the JSON content to the external processor:
- save it to a temp file and give it as input to your processor
- modify `RunProcess.execute` to take as input a OutputStream. This link seems helpful: https://stackoverflow.com/questions/18903549/writing-to-inputstream-of-a-java-process

I would prefer the first option though since it leaves room for debugging. I could inspect the JSON file and look for errors when they will happen.

Reply:
I went with the temp file solution, and indeed having it present to inspect was nice.